libres/lib/res_util/subst_list.cpp

```diff
@@ -785,62 +785,113 @@ void subst_list_fprintf(const subst_list_type * subst_list , FILE * stream) {
 }
 
 
-/** Will loose tagging .... */
-int subst_list_add_from_string( subst_list_type * subst_list , const char * arg_string, bool append) {
-  int     error_count = 0;
-  if (arg_string != NULL) {
-    char ** key_value_list;
-    int     num_arg, iarg;
-
-    util_split_string(arg_string , "," , &num_arg , &key_value_list);
-    for (iarg = 0; iarg < num_arg; iarg++) {
-      if (strchr(key_value_list[iarg] , '=') == NULL)
-        //util_abort("%s: could not find \'=\' in argument string:%s \n",__func__ , key_value_list[iarg]);
-        /*
-          Could not find '=' in the argument string, this argument will
-          be ignored, and the error_count will be increased by one.
-        */
-        error_count += 1;
-      else {
-        char * key , * value;
-        char * tmp     = key_value_list[iarg];
-        int arg_length , value_length;
-        while (isspace(*tmp))  /* Skipping initial space */
-          tmp++;
-
-        arg_length = strcspn(tmp , " =");
-        key  = util_alloc_substring_copy(tmp , 0 , arg_length);
-        tmp += arg_length;
-        while ((*tmp == ' ') || (*tmp == '='))
-          tmp++;
-
-        value_length = strcspn(tmp , " ");
-        value = util_alloc_substring_copy( tmp , 0 , value_length);
-
-        /* Setting the argument */
+static int find_substring(const char *arg_string, const char *split_char)
+{
+    char pattern[4] = {'"', '\'', '\0', '\0'};
+    strcat(pattern, split_char);
+    int len = strcspn(arg_string, pattern);
+
+    while (strlen(arg_string) > len && (arg_string[len] == '"' || arg_string[len] == '\''))
+    {
+        const char delimiter = arg_string[len];
+
+        // Add the delimiter.
+        len++;
+
+        // The string must be long enough to accomdate a corresponding delimiter.
+        if (strlen(arg_string) <= len)
+            return -1;
+
+        // Find the corresponding delimiter.
+        const char *end = strchr(arg_string + len + 1, delimiter);
+
+        // No corresponding end delimiter is an error.
+        if (end == NULL)
+            return -1;
+
+        len = end - arg_string + 1;
+
+        // Keep searching if there is more.
+        if (strlen(end + 1) > 0)
+            len += strcspn(end + 1, pattern);
+    }
+
+    return len;
+}
+
+static char *strip_string(char *str)
+{
+    while (isspace(*str))
+        str++;
+    int len = strlen(str);
+    while (len > 0 && isspace(str[len - 1]))
+        len--;
+    str[len] = '\0';
+    return str;
+}
```

Comment:
🤭 http://www.and.org/vstr/comparison

Reply:
Add comment on what this does:

- substring
- trailing and leading

Reply:
String handling with the plain C standard library is actually not too bad, but of course there are plenty of libraries to do that nicer. But I don't think we should pull in another library just for that purpose.

 In fact I considered rewriting the function in C++, which would probably much easier, or at least easier to read. However, since this code was already C, I decided to stick with it. 

But I just realized that there are some utility functions in libres/libecl anyway that could be useful. I will look into that.